iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -47,13 +48,17 @@ class ServiceDiscovery_test : public RouDi_GTest
   public:
     iox::runtime::PoshRuntime* runtime{&iox::runtime::PoshRuntime::initRuntime("Runtime")};
     ServiceDiscovery sut;
-    static bool callbackWasCalled;
+    static std::atomic_bool callbackWasCalled;
     static ServiceContainer serviceContainer;
 
     void SetUp()
     {
         callbackWasCalled = false;
         serviceContainer.clear();
+        m_watchdog.watchAndActOnFailure([] {
+            EXPECT_TRUE(false);
```

Comment:
Could use `FAIL();`. Do we really need to call terminate? Only needed if we cannot recover and run further tests. 

We may have a thread in deadlock at this point (we cannot know by waiting a finite time), so we probably need to. There always can be an arbitrary error in implementation causing a deadlock (but this is the case in any test and we do not use a watchdog every time ...). If we are sure that we can resolve the deadlock by e.g. destroying the sut we would not have to terminate.

Resolve me.